docs/java-rest/high-level/ml/update-datafeed.asciidoc

```diff
@@ -0,0 +1,95 @@
+--
+:api: update-datafeed
+:request: UpdateDatafeedRequest
+:response: PutDatafeedResponse
+--
+[id="{upid}-{api}"]
+=== Update Datafeed API
+
+The Update Datafeed API can be used to update a {ml} datafeed
+in the cluster. The API accepts a +{request}+ object
+as a request and returns a +{response}+.
+
+[id="{upid}-{api}-request"]
+==== Update Datafeed Request
+
+A +{request}+ requires the following argument:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests-file}[{api}-request]
+--------------------------------------------------
+<1> The updated configuration of the {ml} datafeed
+
+[id="{upid}-{api}-config"]
+==== Updated Datafeed Configuration
```

Comment:
This title/section might be unnecessary. Check `update-job` documentation. We could have a similar structure here unless you dislike that one.